File: Python/Arrays and Hashing/LongestConsecutiveSequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        best = 0
        seq = 0
        nums = set(nums)  #O(n) operation

        for num in nums:
            if (num-1) not in nums:     #imp one to make sure already counted ones are not included
                seq = num + 1;
                while seq in nums:
                    seq += 1;
                best = max(best, seq-num)

        return best

File: C++/Arrays and Hashing/LongestConsecutiveSequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s; //Here I used set, which gave a TLE, make sure you use unordered one
        for(auto num:nums){
            s.insert(num);
            //cout << num << " ";
        }

        int ans = 0 , seq =0 ;

        for(auto num : nums){
            if(s.find(num-1)==s.end()){ //element not in previous sequences
                seq = num +1;
                //cout << num << " ";
                while(s.find(seq)!=s.end()){
                    seq++;
                }
                ans = max(ans, seq-num);
            }
        }
        return ans;
    }
};